test/external/dataplane/connectivity.go

```diff
@@ -38,179 +38,216 @@ const (
+		// Access from a hostnetwork pod to external apps is not supported
+	})
 
-	// Get handle for existing docker.
-	docker := framework.New(extAppName)
-	dockerIP := docker.GetIP(extNetName)
+	When("connected from a non-gateway-cluster", func() {
+		BeforeEach(func() {
+			cluster, err = getNonGatewayClusterIndex(framework.TestContext.ClusterIDs)
+			Expect(err).NotTo(HaveOccurred())
+		})
 
-	// Create service without selector and endpoints for dockerIP, and export the service.
-	extSvc := f.CreateTCPServiceWithoutSelector(extClusterIdx, "extsvc", "http", 80)
-	f.CreateTCPEndpoints(extClusterIdx, extSvc.Name, "http", dockerIP, 80)
-	f.CreateServiceExport(extClusterIdx, extSvc.Name)
+		When("a pod connects via TCP to a remote pod", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.PodIP
+				networking = framework.PodNetworking
+			})
 
-	// Get globalIPs for the extApp to use later.
-	extIngressGlobalIP := f.AwaitGlobalIngressIP(extClusterIdx, extSvc.Name, extSvc.Namespace)
-	Expect(extIngressGlobalIP).ToNot(Equal(""))
+			When("the pod is not on a gateway", func() {
+				verifyInteraction(framework.NonGatewayNode)
+			})
 
-	extEgressGlobalIPs := f.AwaitClusterGlobalEgressIPs(extClusterIdx, constants.ClusterGlobalEgressIPName)
-	Expect(extEgressGlobalIPs).ToNot(BeEmpty())
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
 
-	for idx := range framework.KubeClients {
-		clusterName := framework.TestContext.ClusterIDs[idx]
+		When("a pod connects via TCP to a remote service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.ServiceIP
+				networking = framework.PodNetworking
+			})
 
-		By(fmt.Sprintf("Creating a pod and a service in cluster %q", clusterName))
+			When("the pod is not on a gateway", func() {
+				verifyInteraction(framework.NonGatewayNode)
+			})
 
-		np := f.NewNetworkPod(&framework.NetworkPodConfig{
-			Type:    framework.CustomPod,
-			Port:    80,
-			Cluster: framework.ClusterIndex(idx),
-			// Also test NonGatewayNode
-			Scheduling:    framework.GatewayNode,
-			ContainerName: testContainerName,
-			ImageName:     testImage,
-			Command:       simpleHTTPServerCommand,
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
 		})
-		svc := np.CreateService()
-		f.CreateServiceExport(np.Config.Cluster, svc.Name)
 
-		// Get globalIPs for the network pod to use later.
-		remoteIP := f.AwaitGlobalIngressIP(np.Config.Cluster, svc.Name, svc.Namespace)
-		Expect(remoteIP).ToNot(Equal(""))
+		// Access from a hostnetwork pod to external apps is not supported
+	})
+})
 
-		podGlobalIPs := f.AwaitClusterGlobalEgressIPs(np.Config.Cluster, constants.ClusterGlobalEgressIPName)
-		Expect(podGlobalIPs).ToNot(BeEmpty())
+func testExternalConnectivity(p testParams) {
+	externalClusterName := getGatewayClusterName(framework.TestContext.ClusterIDs)
```

Comment:
```suggestion
	gateway-cluster := getGatewayClusterName(framework.TestContext.ClusterIDs)
```

To match with the topology, lets use gateway-cluster instead of externalCluster.

Reply:
Changed to `gatewayCluster`, but we may change the name again.